File: flutter_app/lib/features/customers/models/customer.dart
import 'package:flutter/foundation.dart';

@immutable
class Customer {
  const Customer({
    required this.id,
    required this.name,
    required this.phone,
    this.email,
    required this.points,
    required this.totalSpent,
    required this.visitCount,
    this.lastVisit,
    required this.createdAt,
  });

  final String id;
  final String name;
  final String phone;
  final String? email;
  final int points;
  final int totalSpent;
  final int visitCount;
  final DateTime? lastVisit;
  final DateTime createdAt;

  factory Customer.fromJson(Map<String, dynamic> json) => Customer(
        id: json['id'] as String,
        name: json['name'] as String,
        phone: json['phone'] as String,
        email: json['email'] as String?,
        points: (json['points'] as num?)?.toInt() ?? 0,
        totalSpent: (json['total_spent'] as num?)?.toInt() ?? 0,
        visitCount: (json['visit_count'] as num?)?.toInt() ?? 0,
        lastVisit: json['last_visit'] != null
            ? DateTime.parse(json['last_visit'] as String)
            : null,
        createdAt: DateTime.parse(json['created_at'] as String),
      );

  Map<String, dynamic> toJson() => {
        'name': name,
        'phone': phone,
        if (email != null) 'email': email,
      };
}

File: backend/internal/models/customer.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Customer struct {
	ID         uuid.UUID  `json:"id"`
	Name       string     `json:"name"`
	Phone      string     `json:"phone"`
	Email      *string    `json:"email,omitempty"`
	Points     int        `json:"points"`
	TotalSpent int64      `json:"total_spent"`
	VisitCount int        `json:"visit_count"`
	LastVisit  *time.Time `json:"last_visit,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`
}

type CreateCustomerRequest struct {
	Name  string  `json:"name"`
	Phone string  `json:"phone"`
	Email *string `json:"email,omitempty"`
}

type UpdateCustomerRequest struct {
	Name   *string `json:"name,omitempty"`
	Phone  *string `json:"phone,omitempty"`
	Email  *string `json:"email,omitempty"`
	Points *int    `json:"points,omitempty"`
}
